src/core/_config.scss

```diff
@@ -37,6 +38,160 @@ $shadow-color-regular: hsla(0, 0, 0%, 0.2) !default;
+$dark-warning: #ddcd45 !default;
+$dark-danger: #ff8c86 !default;
+$dark-muted: #868e96 !default;
+
+$dark-primary-light: lighten($dark-primary, 50%) !default;
+$dark-secondary-light: lighten($dark-secondary, 50%) !default;
+$dark-success-light: lighten($dark-success, 30%) !default;
+$dark-warning-light: lighten($dark-warning, 30%) !default;
+$dark-danger-light: lighten($dark-danger, 45%) !default;
+$dark-muted-light: lighten($dark-muted, 35%) !default;
+
+$dark-white-dark: rgba(#000, 0.03) !default;
+$dark-light-dark: rgba(#000, 0.7) !default;
+$dark-white: rgba(#fff, 1) !default;
+$dark-main-background: rgba(#41403e, 1) !default;
+$dark-main-background-light: lighten($dark-main-background, 50%) !default;
+
+$dark-primary-text: #41403e !default;
+$dark-secondary-text: $dark-secondary !default;
+$dark-success-text: $dark-primary !default;
+$dark-warning-text: $dark-primary !default;
+$dark-danger-text: $dark-primary !default;
+$dark-muted-text: $dark-primary !default;
+
+$dark-shadow-color-regular: hsla(0, 0, 0%, 0.2) !default;
+$dark-shadow-color-hover: hsla(0, 0, 0%, 0.3) !default;
+
+
+
+$theme-map: (
+  light: (
+    primary: $primary,
+    secondary: $secondary,
+    success: $success,
+    warning: $warning,
+    danger: $danger,
+    muted: $muted,
+    primary-light: #{$primary-light},
+    secondary-light:$secondary-light,
+    success-light:$success-light,
+    warning-light:$warning-light,
+    danger-light:$danger-light,
+    muted-light:$muted-light,
+    white-dark:$white-dark,
+    light-dark:$light-dark,
+    white:$white,
+    main-background: $main-background,
+    main-background-light: $main-background-light,
+    primary-text:$primary-text,
+    secondary-text:$secondary-text,
+    success-text:$success-text,
+    warning-text:$warning-text,
+    danger-text:$danger-text,
+    muted-text:$muted-text,
+    shadow-color-regular: $shadow-color-regular,
+    shadow-color-hover: $shadow-color-hover,
+  ),
+  dark: (
+    primary: $dark-primary,
+    secondary: $dark-secondary,
+    success: $dark-success,
+    warning: $dark-warning,
+    danger: $dark-danger,
+    muted: $dark-muted,
+    primary-light: #{$dark-primary-light},
+    secondary-light:$dark-secondary-light,
+    success-light:$dark-success-light,
+    warning-light:$dark-warning-light,
+    danger-light:$dark-danger-light,
+    muted-light:$dark-muted-light,
+    white-dark:$dark-white-dark,
+    light-dark:$dark-light-dark,
+    white:$dark-white,
+    main-background: $dark-main-background,
+    main-background-light: $dark-main-background-light,
+    primary-text:$dark-primary-text,
+    secondary-text:$dark-secondary-text,
+    success-text:$dark-success-text,
+    warning-text:$dark-warning-text,
+    danger-text:$dark-danger-text,
+    muted-text:$dark-muted-text,
+    shadow-color-regular: $dark-shadow-color-regular,
+    shadow-color-hover: $dark-shadow-color-hover,
+  )
+);
+
+
+html {
+  --primary: #{map-get(map-get($theme-map, 'light'), 'primary')};
```

Comment:
Is there a way to map over the theme to generate these values to generate this dynamically? It might look something like this, but my Sass is a little rusty

```scss
@each $theme, $colors in $theme-map {
  @each $name, $color in $colors {
    --#{@name}: $color;
  }
}
```

Reply:
That's a great idea, thanks!